subsys/mgmt/ec_host_cmd/backends/ec_host_cmd_backend_usb.c

```diff
@@ -0,0 +1,674 @@
+	if (USB_EP_DIR_IS_OUT(ep)) {
+		buf->len = 0;
+	}
+
+	return buf;
+}
+
+static struct net_buf *ec_host_cmd_buf_alloc_int(const uint8_t ep)
+{
+	struct net_buf *buf = NULL;
+	struct udc_buf_info *bi;
+
+	buf = net_buf_alloc(&int_ep_pool, K_NO_WAIT);
+	if (!buf) {
+		return NULL;
+	}
+
+	bi = udc_get_buf_info(buf);
+	bi->ep = ep;
+
+	return buf;
+}
+
+static int ec_host_cmd_signal_event(struct usbd_class_data *c_data)
+{
+	struct usbd_context *uds_ctx = usbd_class_get_ctx(c_data);
+	struct ec_host_cmd_usb_ctx *ctx = usbd_class_get_private(c_data);
+	struct net_buf *buf;
+	int ret;
+
+	buf = ec_host_cmd_buf_alloc_int(ec_host_cmd_get_in_int_ep(c_data));
+	if (buf == NULL) {
+		LOG_WRN("Event has been already signaled but not polled by the host");
+		return 0;
+	}
+	net_buf_add_u8(buf, USB_EC_HOST_CMD_IRQ_TYPE_EVENT);
+	buf->len = EP_INT_SIZE_SIZE;
+	ret = usbd_ep_enqueue(c_data, buf);
+	if (ret) {
+		LOG_ERR("Failed to enqueue EP IN INT: %d", ret);
+		usbd_ep_buf_free(uds_ctx, buf);
+		return ret;
+	}
+	ctx->pending_event = false;
+
+	return 0;
+}
+
+static int handle_out_transfer(struct usbd_class_data *c_data)
+{
+	int ret;
+	struct ec_host_cmd_usb_ctx *ctx = usbd_class_get_private(c_data);
+	size_t expected_len =
+		expected_request_len((struct ec_host_cmd_request_header *)ctx->rx_ctx->buf);
+
+	/* Notify about a new command or response with a proper error code. */
+	if ((ctx->rx_ctx->len >= expected_len) || (expected_len == 0) ||
+	    (expected_len > ctx->rx_ctx->len_max)) {
+		k_work_cancel_delayable(&ctx->reset_work);
+		if (ctx->rx_ctx->len > expected_len) {
+			LOG_ERR("Received incorrect number of bytes, got: %d, expected: %d",
+				ctx->rx_ctx->len, expected_len);
+		}
+		ctx->state = USB_EC_HOST_CMD_STATE_PROCESSING;
+		ec_host_cmd_rx_notify();
+	}
+
+	/* Enqueue OUT transfer if we are still receiving. */
+	if (ctx->state == USB_EC_HOST_CMD_STATE_RECEIVING) {
+		if (!IS_UDC_ALIGNED(ctx->rx_ctx->len)) {
+			LOG_ERR("Received unaligned OUT transfer: %d", ctx->rx_ctx->len);
+			k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+			return 0;
+		}
+		ctx->usb_rx_buf = ec_host_cmd_buf_alloc(
+			ec_host_cmd_get_out_ep(c_data), expected_len - ctx->rx_ctx->len,
+			ctx->rx_ctx->buf + ctx->rx_ctx->len);
+		if (ctx->usb_rx_buf == NULL) {
+			LOG_ERR("Failed to allocate buf OUT");
+			k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+			return 0;
+		}
+		ret = usbd_ep_enqueue(c_data, ctx->usb_rx_buf);
+		if (ret) {
+			net_buf_unref(ctx->usb_rx_buf);
+			k_work_reschedule(&ctx->reset_work, K_NO_WAIT);
+			LOG_ERR("Failed to enqueue EP OUT: %d", ret);
+			return 0;
+		}
+	}
+
+	return 0;
+}
+
+static int ec_host_cmd_request(struct usbd_class_data *c_data, struct net_buf *buf, int err)
+{
+	struct ec_host_cmd_usb_ctx *ctx = usbd_class_get_private(c_data);
+	struct usbd_context *uds_ctx = usbd_class_get_ctx(c_data);
+	struct udc_buf_info *bi = NULL;
+	int ret;
+
+	bi = udc_get_buf_info(buf);
+
+	if (err) {
+		LOG_ERR("EP: 0x%02x, err: %d", bi->ep, err);
```

Comment:
Maybe you do not want to log cancelled request, that could confuse your coworkers.

https://github.com/zephyrproject-rtos/zephyr/blob/789f3cea8c17792077774c974d01546896dd80df/subsys/usb/device_next/class/usbd_cdc_acm.c#L281-L289

Reply:
Makes sense, modified the logging.